lib/dependency-manager-adapters/npm.js

```diff
@@ -147,14 +147,16 @@ module.exports = CoreObject.extend({
     debug('Changing to dependency set: %s', JSON.stringify(depSet));
 
     if (!depSet) { return RSVP.resolve([]); }
-    let backupPackageJSON = path.join(this.cwd, this.packageJSONBackupFileName);
-    let packageJSONFile = path.join(this.cwd, this.packageJSON);
-    let packageJSON = JSON.parse(fs.readFileSync(backupPackageJSON));
-    let newPackageJSON = this._packageJSONForDependencySet(packageJSON, depSet);
 
-    debug('Write package.json with: \n', JSON.stringify(newPackageJSON));
+    return this._restoreOriginalDependencies().then(() => {
```

Comment:
It seems very bizarre to restore before applying the depSet changes here.

Any cleanup/restore that was supposed to happen should have happened **before** the next scenario runs it's `applyDependencySet`.